tools/metricdoctool/metricdoctool.go

```diff
@@ -0,0 +1,369 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/cilium/cilium/daemon/cmd"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+func main() {
+	agent := generateMetricDocForCells(cmd.Agent)
+
+	tpl, err := template.New("page").Parse(pageTemplate)
+	if err != nil {
+		panic(err)
+	}
+	tpl.Execute(os.Stdout, struct {
+		Agent string
+	}{
+		Agent: agent,
+	})
+}
+
+//go:embed template/page.gotmpl
+var pageTemplate string
+
+type metricsIn struct {
+	cell.In
+
+	Metrics []metric.WithMetadata `group:"hive-metrics"`
+}
+
+func generateMetricDocForCells(cells ...cell.Cell) string {
+	var agentMetrics []metric.WithMetadata
+
+	cells = append(cells, cell.Invoke(func(min metricsIn) {
+		agentMetrics = min.Metrics
+	}))
+
+	// Collect all metrics which are provided by the agent and the cells it depends on
+	h := hive.New(
+		cells...,
+	)
+	h.Populate()
+
+	metricPerSubsystem := make(map[string]perSubsystem)
+	for _, m := range agentMetrics {
+		subsys := m.Opts().Subsystem
+		key := subsys.Name
+
+		// The metric truly has no category if no subsystem name and doc-name is set
+		if subsys.Name == "" {
+			if subsys.DocName == "" {
+				key = "misc"
+				subsys = metric.Subsystem{
+					Name:    "",
+					DocName: "Misc",
+				}
+			} else {
+				// If we do have a doc-name, then the place it into its own category in the docs, but the metric
+				// name will not be prefixed with anything.
+				key = strings.Replace(strings.ToLower(subsys.DocName), " ", "_", -1)
+			}
+		}
+
+		list := metricPerSubsystem[key]
+		list.Subsystem = subsys
+		list.Metrics = append(list.Metrics, m)
+		list.Labels = make(map[string]metric.LabelDescription)
+
+		metricPerSubsystem[key] = list
+	}
+
+	// Merge all label definitions per subsystem
+	for _, mps := range metricPerSubsystem {
+		for _, m := range mps.Metrics {
+			lbls := m.Labels()
+			for _, lbl := range lbls {
+				desc := mps.Labels[lbl.Name]
+				desc.Name = lbl.Name
+
+				// If there are multiple descriptions for a label, pick the longest. If two are the same size, pick the
+				// alphabetically first. This should ensure a consistent outcome
```

Comment:
Edits for clarity, consistent periods, avoiding soft conditionals
```suggestion
				// If there are multiple descriptions for a label, pick the longest. If two are the same size, pick the
				// first in alphabetical order. This ensures a consistent outcome.
```